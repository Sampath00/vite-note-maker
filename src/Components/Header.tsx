import { CopyIcon, EditIcon } from '@chakra-ui/icons';
import { Button, Flex, HStack, Text } from '@chakra-ui/react';
import * as React from 'react';
import { useState } from 'react';
import { Link } from 'react-router-dom';


interface IHeaderProps {
}

const Header: React.FunctionComponent<IHeaderProps> = () => {
    const[flag,setFlag] = useState(true);
  return(
    <Flex as="nav" p='10px' bg='gray.300'>
        <HStack spacing="1095px">
            <Text fontSize='2xl' fontWeight='semibold'> Note Maker </Text>
            {flag ?
                <Link to="/create">
                    <Button colorScheme='teal' onClick={()=>setFlag(!flag)} leftIcon={<EditIcon/>}> Create </Button>
                </Link>
            :
                <Link to="/">
                    <Button colorScheme='linkedin' onClick={()=>setFlag(!flag)} leftIcon={<CopyIcon/>}> Notes </Button>
                </Link>
            }
            
        </HStack>
    </Flex>
  ) ;
};

export default Header;
