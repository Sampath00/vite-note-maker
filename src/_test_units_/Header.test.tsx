import {describe, test, expect, beforeEach} from "vitest"
import Header from "../Components/Header"
import { render, screen} from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

describe("Testing on Header Component", () => {

    beforeEach(() => {
        render(<MemoryRouter><Header/></MemoryRouter>)
    })

    test("Header component text", () => {
        expect(screen.getByText(/note maker/i)).toBeDefined();
    })

    test("Header snapshot test",()=>{
        const wrapper = render(<MemoryRouter><Header/></MemoryRouter>);
        expect(wrapper).toMatchSnapshot()
    })
})