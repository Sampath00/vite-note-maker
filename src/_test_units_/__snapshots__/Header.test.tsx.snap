// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing on Header Component > Header snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="css-d9w2cm"
      >
        <div
          class="chakra-stack css-wdzbn0"
        >
          <p
            class="chakra-text css-37kvc8"
          >
             Note Maker 
          </p>
          <a
            href="/create"
          >
            <button
              class="chakra-button css-4xx2wk"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-width="2"
                  >
                    <path
                      d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                    />
                    <path
                      d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                    />
                  </g>
                </svg>
              </span>
               Create 
            </button>
          </a>
        </div>
      </nav>
    </div>
    <div>
      <nav
        class="css-d9w2cm"
      >
        <div
          class="chakra-stack css-wdzbn0"
        >
          <p
            class="chakra-text css-37kvc8"
          >
             Note Maker 
          </p>
          <a
            href="/create"
          >
            <button
              class="chakra-button css-4xx2wk"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-width="2"
                  >
                    <path
                      d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                    />
                    <path
                      d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                    />
                  </g>
                </svg>
              </span>
               Create 
            </button>
          </a>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="css-d9w2cm"
    >
      <div
        class="chakra-stack css-wdzbn0"
      >
        <p
          class="chakra-text css-37kvc8"
        >
           Note Maker 
        </p>
        <a
          href="/create"
        >
          <button
            class="chakra-button css-4xx2wk"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="2"
                >
                  <path
                    d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                  />
                  <path
                    d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                  />
                </g>
              </svg>
            </span>
             Create 
          </button>
        </a>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
